internal/gps/project_root_cache.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gps
+
+import "github.com/pkg/errors"
+
+// ProjectRootCache caches the result of deducing a project root from a
+// source manager, which is quite helpful when repeatedly converting from
+// import paths to root projects.
+type ProjectRootCache struct {
+	prs map[string]ProjectRoot
```

Comment:
this information should already be cached inside of a `SourceManager` (or, at least, it definitely is in `sourceMgr`) - what's the benefit of layering this on top?